tools/src/test/java/com/nedap/archie/adl14/ADL14ToADL2Test.java

```diff
@@ -0,0 +1,133 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Stack;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class ADL14ToADL2Test {
+
+    @Test
+    public void Adl14ToAdl2IdCodedTest() throws Exception {
+        Archetype adl14;
+        try (InputStream stream = getClass().getResourceAsStream("openEHR-EHR-OBSERVATION.body_weight-adl14.v2.adl")) {
+            ADL14Parser parser = new ADL14Parser(BuiltinReferenceModels.getMetaModels());
+            adl14 = parser.parse(stream, ConversionConfigForTest.getConfig());
+        }
+        Archetype expected;
+        try (InputStream stream = getClass().getResourceAsStream("openEHR-EHR-OBSERVATION.body_weight-adl2.v2.1.8.adls")) {
+            ADLParser parser = new ADLParser(BuiltinReferenceModels.getMetaModels());
+            expected = parser.parse(stream);
+        }
+
+        ADL14ConversionConfiguration configuration = new ADL14ConversionConfiguration();
+        configuration.setAdlConfiguration(ADL14ConversionConfiguration.ADL2VERSION.ID_CODED);
+        ADL14Converter converter = new ADL14Converter(BuiltinReferenceModels.getMetaModels(), configuration);
+
+        ADL2ConversionResultList resultList = converter.convert(List.of(adl14));
+        Archetype result = resultList.getConversionResults().get(0).getArchetype();
+
+        assertEquals("2.3.0", result.getAdlVersion());
+        assertArchetypes(expected, result);
+        assertArchetypes(result, expected); // Also check the reverse, to check if the result doesn't have more than the expected archetype
+    }
+
+    @Test
+    public void Adl14ToAdl2AtCodedTest() throws Exception {
+        Archetype adl14;
+        try (InputStream stream = getClass().getResourceAsStream("openEHR-EHR-OBSERVATION.body_weight-adl14.v2.adl")) {
+            ADL14Parser parser = new ADL14Parser(BuiltinReferenceModels.getMetaModels());
+            adl14 = parser.parse(stream, ConversionConfigForTest.getConfig());
+        }
+        Archetype expected;
+        try (InputStream stream = getClass().getResourceAsStream("openEHR-EHR-OBSERVATION.body_weight-adl2plus.v2.1.8.adls")) {
+            ADLParser parser = new ADLParser(BuiltinReferenceModels.getMetaModels());
+            expected = parser.parse(stream);
+        }
+
+        ADL14ConversionConfiguration configuration = new ADL14ConversionConfiguration();
+        configuration.setAdlConfiguration(ADL14ConversionConfiguration.ADL2VERSION.AT_CODED);
+        ADL14Converter converter = new ADL14Converter(BuiltinReferenceModels.getMetaModels(), configuration);
+
+        ADL2ConversionResultList resultList = converter.convert(List.of(adl14));
+        Archetype result = resultList.getConversionResults().get(0).getArchetype();
+
+        Path path = Paths.get("result-at.adls");
+        Files.write(path, ADLArchetypeSerializer.serialize(result).getBytes(), StandardOpenOption.CREATE, StandardOpenOption.WRITE);
```

Comment:
Should this write action still be here? Or was it just for testing? It might perhaps be useful to keep it in, but commented?